craft_parts/plugins/ant_plugin.py

```diff
@@ -0,0 +1,183 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The Ant plugin."""
+
+import logging
+from typing import Any, Dict, List, Optional, Set, cast
+
+from craft_parts import errors
+
+from . import validator
+from .base import Plugin, PluginModel, extract_plugin_properties
+from .properties import PluginProperties
+
+logger = logging.getLogger(__name__)
+
+
+class AntPluginProperties(PluginProperties, PluginModel):
+    """The part properties used by the Ant plugin."""
+
+    ant_build_targets: List[str] = []
+    ant_build_file: Optional[str] = None
+    ant_properties: Dict[str, str] = {}
+
+    source: str
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]):
+        """Populate make properties from the part specification.
+
+        :param data: A dictionary containing part properties.
+
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = extract_plugin_properties(
+            data, plugin_name="ant", required=["source"]
+        )
+        return cls(**plugin_data)
+
+
+class AntPluginEnvironmentValidator(validator.PluginEnvironmentValidator):
+    """Check the execution environment for the Ant plugin.
+
+    :param part_name: The part whose build environment is being validated.
+    :param env: A string containing the build step environment setup.
+    """
+
+    def validate_environment(self, *, part_dependencies: Optional[List[str]] = None):
+        """Ensure the environment contains dependencies needed by the plugin.
+
+        :param part_dependencies: A list of the parts this part depends on.
+
+        :raises PluginEnvironmentValidationError: If ``ant`` is invalid
+        and there are no parts named ant.
+        """
+        version = self.validate_dependency(
+            dependency="ant",
+            plugin_name="ant",
+            part_dependencies=part_dependencies,
+            argument="-version",
+        )
+        if not version.startswith("Apache Ant") and (
```

Comment:
It's a good thing that ant doesn't use control characters for terminal formatting to display the version like maven does.